cmd/kube-apiserver/app/server.go

```diff
@@ -367,6 +367,8 @@ func CreateKubeAPIServerConfig(
 			ServiceAccountMaxExpiration: s.ServiceAccountTokenMaxExpiration,
 
 			VersionedInformers: versionedInformers,
+
+			CloudProvider: s.ServerRunOptions.CloudProvider,
```

Comment:
Please do not do this. 
The cloud provider extraction effort of sig cloud-provider is trying to break all the cloud provider dependencies within core Kubernetes components.

Reply:
If `ServerRunOptions` cannot be passed through, `register` doesn't know whether ccm exists. Is there any better solution? 🤔

Reply:
I don't object to passing through the ServerRunOptions but the CloudProvider struct should be deprecated and removed from it soon. If we decide we want to do this we might be better of providing a more generic mechanism such as providing a list of external services and health endpoints we can call. However I would also look at https://github.com/kubernetes/enhancements/issues/553 as it details a lot of issues with component status and a desire to deprecate it.

Reply:
Also worth looking at https://github.com/kubernetes/kubernetes/issues/17461 and https://github.com/kubernetes/kubernetes/pull/13216.